Added extra context in the nested string interpolation example

ellyxir opened the pull request:
the string + operator was not previously introduced and it wasn't clear that you can have more than just a variable within ${}
im doing a video series for learning the nix programming language and when i saw this part of the language tutorial, i had a hard time parsing what i was seeing. perhaps this could make it more clear. thank you!

fricklerhandwerk commented:
Yeah the tutorial is deliberately not going into depth about operators, because they're largely unsurprising -- except of course for string/path concatenation.

Thanks for pointing it out.

Instead of prefixing the note,
I suggest moving it into a collapsible "detailed explanation" block like it's done elsewhere, and link to the right place in the Nix manual for details on that particular operator.

Could you please do that? If not, I'll keep a reminder to finish it up.

ellyxir commented:
Yes I'd be happy to but won't get to it until Monday. Hope that's ok!

ellyxir commented:
I've moved it to a detailed explanation block and added a little more detail. Also linked to manual and the functional libraries section because standard operators and functions are mentioned there. Let me know if this is ok. Thank you!

fricklerhandwerk commented on source/tutorials/nix-language.md:
````diff
@@ -822,6 +822,26 @@ in
 ```
 
+:::{dropdown} Detailed explanation
+The `+` sign in the above expression is the concatenation operator. It is one of the many [built-in operators][operators] (`+`, `==`, `&&`, etc.) These operators can be used within `${ }`.
````
```suggestion
Any Nix expression where the value can be represented as a string can be used within `${ }`.

The `+` sign in the above expression is the [string concatenation operator](https://nix.dev/manual/nix/latest/language/operators#string-concatenation), which takes two strings and produces a new string.

The expression in the example is deliberately confusing in order to demonstrate that arbitrarily nested string interpolations are possible, but tend to be hard to read.

It denotes a string that contains the interpolation of concatenating the value of `a` with a string that starts with a space and is followed by another interpolated string.
That second interpolated string is again the result of concatenating the value of `a` and yet another string that starts with a space and is followed by an interpolation of `a`.
```

fricklerhandwerk reviewed (approved):
Thanks a lot @ellyxir. This prompted me to fix [a small discoverability issue](https://github.com/NixOS/nix/pull/11793) in the Nix manual.

I expanded a bit on the explanation.
